tauri-updater/README.md

```diff
@@ -0,0 +1,275 @@
+			"url":"https://github.com/lemarier/tauri-test/releases/download/v1.0.0/app.x64.msi.zip"
+		}
+	}
+}
+```
+
+
+# Bundler (Artifacts)
+
+The Tauri bundler will automatically generate update artifacts if the updater is enabled in your tauri.config.js
+
+If the bundler can locate your private and pubkey, your update artifacts will be automatically signed.
+
+The signature can be found in the `sig` file. The signature can be uploaded to GitHub safely or made public as long as your private key is secure.
+
+You can see how it's [bundled with the CI](https://github.com/tauri-apps/tauri/blob/feature/new_updater/.github/workflows/artifacts-updater.yml#L44)
+
+## macOS
+
+On MACOS we create a .tar.gz from the whole application. (.app)
+
+```
+target/release/bundle
+└── osx
+    └── app.app
+    └── app.app.tar.gz (update bundle)
+    └── app.app.tar.gz.sig (if signature enabled)
+```
+
+## Windows
+
+On Windows we create a .zip from the MSI, when downloaded and validated, we run the MSI install.
+
+```
+target/release
+└── app.x64.msi
+└── app.x64.msi.zip (update bundle)
+└── app.x64.msi.zip.sig (if signature enabled)
+```
+
+## Linux
+
+On Linux, we create a .tar.gz from the AppImage.
+
+```
+target/release/bundle
+└── appimage
+    └── app.AppImage
+    └── app.AppImage.tar.gz (update bundle)
+    └── app.AppImage.tar.gz.sig (if signature enabled)
+```
+
+# Signing updates
+
+We offer a built-in signature to ensure your update is safe to be installed.
+
+To sign your updates, you need two things.
+
+The *Public-key* (pubkey) should be added inside your tauri.config.js to validate the update archive before installing.
+
+The *Private key* (privkey) is used to sign your update and should NEVER be shared with anyone. Also, if you lost this key, you'll NOT be able to publish a new update to the current user base (if pubkey is set in tauri.config.js). It's important to save it at a safe place and you can always access it.
+
+To generate your keys you need to use the Tauri cli. (will be built in Tauri main CLI soon)
+
+```bash
+cargo tauri-sign-updates -g
+```
+
+You have multiple options available
+```bash
+OPTIONS:
+    -f, --force                              Overwrite private key even if exist on the specified path
+    -g, --generate-key                       Generate keypair to sign files
+    -h, --help                               Prints help information
+        --no-password                        Set empty password for your private key
+    -P, --password <PASSWORD>                Set private key password when signing
+        --private-key <STRING>               Load the private key from a string
+    -p, --private-key-path <PATH>            Load the private key from a file
+    -s, --sign <PATH>                        Sign the specified binary
+    -w, --write-private-key <DESTINATION>    Write private key to a file
+```
+
+***
+
+Environment variables used to sign:
+
+`TAURI_PRIVATE_KEY`  Path or String of your private key
```

Comment:
How does the CLI tell whether the value of this variable is a path or the private key itself?

Reply:
You can see it here:
https://github.com/tauri-apps/tauri/blob/feature/new_updater/cli/tauri-bundler/src/bundle/updater_bundle.rs#L42

It check if the file exist if not it expect to be a string

Reply:
Ah whoops, not sure how I missed that